code/modules/mob/living/simple_animal/friendly/farm_animals.dm

```diff
@@ -354,6 +354,89 @@
 	else
 		STOP_PROCESSING(SSobj, src)
 
+/mob/living/simple_animal/sheep
+	name = "sheep"
+	desc = "It's so fluffy!"
+	icon_state = "sheep"
+	icon_living = "sheep"
+	icon_dead = "sheep_dead"
+	gender = FEMALE
+	mob_biotypes = list(MOB_ORGANIC, MOB_BEAST)
+	speak = list("baa?","baa","BAAAAAA")
+	speak_emote = list("bleats")
+	emote_hear = list("brays.")
+	emote_see = list("nibbles at the ground.")
+	speak_chance = 1
+	turns_per_move = 5
+	see_in_dark = 6
+	butcher_results = list(/obj/item/reagent_containers/food/snacks/meat/slab = 4)
+	response_help_continuous  = "pets"
+	response_help_simple = "pet"
+	response_disarm_continuous = "gently pushes aside"
+	response_disarm_simple = "gently push aside"
+	response_harm_continuous = "kicks"
+	response_harm_simple = "kick"
+	attack_verb_continuous = "kicks"
+	attack_verb_simple = "kick"
+	attack_sound = 'sound/weapons/punch1.ogg'
+	health = 40
+	maxHealth = 40
+	var/obj/item/udder/udder = null
+	var/shaved = FALSE
+	gold_core_spawnable = FRIENDLY_SPAWN
+
+/mob/living/simple_animal/sheep/Initialize()
+	udder = new()
+	. = ..()
+
+/mob/living/simple_animal/sheep/Destroy()
+	QDEL_NULL(udder)
+	return ..()
+
+/mob/living/simple_animal/sheep/attackby(obj/item/O, mob/user, params)
+	if(stat == CONSCIOUS && istype(O, /obj/item/reagent_containers/glass))
+		udder.milkAnimal(O, user)
+		return TRUE
+	else
+		return ..()
+
+/mob/living/simple_animal/sheep/Life()
+	. = ..()
```

Comment:
```suggestion
/mob/living/simple_animal/sheep/BiologicalLife()
	if(!(. = ..()))
		return
```

Reply:
^ the above will also make it so you do'nt need the stat == CONSCIOUS check, as biologicallife returns false if the thing is dead or deleted

Reply:
`if(!(. = ..()))` looks like something out of a loosely esoteric programming language though.